upgrade/scripts/upgrade/util/postgres-reinit.sh

```diff
@@ -0,0 +1,174 @@
+#   may get stuck on pod trying to update its own label during CSM 1.6 upgrade (CASMINST-6728). Is this still the case?
+# - Use "patronictl reinit" command seem to be able to work around the issue
+#
+
+function postgres_operator_running() {
+  if [ "$(kubectl -n services get pod -l app.kubernetes.io/name=postgres-operator --no-headers | awk '{ print $2 ":" $3 }')" == "2/2:Running" ]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+function postgres_pods_running() {
+  if [ "$(kubectl get pod -l application=spilo -A --no-headers | awk '{ print $3 ":" $4 }' | sort -u)" == "3/3:Running" ]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+function postgres_clusters_running() {
+  if [ "$(kubectl get postgresql -A -o json | jq -r '.items[].status.PostgresClusterStatus' | sort -u)" == "Running" ]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+function wait_for() {
+  local command="${1}"
+  local message="${2}"
+  local count=0
+  local total=120
+  local sleep=5
+  while true; do
+    if ${command}; then
+      echo "${message}" | awk '{print toupper(substr($0, 1, 1)) substr($0, 2)}'
+      break
+    else
+      if [ "${count}" -ge "${total}" ]; then
+        echo "ERROR: giving up for ${message} after ${total} attempts"
+        exit 1
+      fi
+      count=$((count + 1))
+      echo "Waiting for ${message}, sleeping for ${sleep} seconds and retry, attempt ${count}/${total} ..."
+      sleep ${sleep}
+    fi
+  done
+}
+
+wait_for postgres_operator_running "postgres-operator pod running"
+# Only recover postgres clusters in argo, services, spire namespaces - but wait for all clusters (such as sma) to be healthy, as
+# preflight checks at the end of prerequisites script will need all clusters anyway.
+#while IFS=: read -r pg_ns pg_cluster; do
+for pg_line in $(kubectl get sts -A -l application=spilo -o json | jq -r '.items[] | select(.metadata.namespace == "argo" or .metadata.namespace == "services" or .metadata.namespace == "spire") | (.metadata.namespace + ":" + .metadata.name)'); do
+  pg_cluster=${pg_line#*:}
+  pg_ns=${pg_line%:*}
+  echo "Working on ${pg_cluster} in namespace ${pg_ns} ..."
+  count=1
+  total=3
+  wait=60
+  pg_pod=""
+  leader=""
+  while [ ${count} -le ${total} ]; do
+    # find one pod from the cluster to run patroni command
+    if kubectl get pods -n ${pg_ns} -l application=spilo --no-headers | grep ${pg_cluster} | grep "3/3" | grep -m 1 Running; then
+      pg_pod=$(kubectl get pods -n ${pg_ns} -l application=spilo --no-headers | grep ${pg_cluster} | grep "3/3" | grep -m 1 Running | awk '{print $1}')
+      echo "INFO: found a running pod ${pg_pod}"
+      break
+    else
+      echo "WARNING: No fully running pod is found for ${pg_cluster} in ${pg_ns}. Rollout restart it and sleep for ${wait} seconds, attempt ${count}/${total})..."
+      kubectl rollout restart sts ${pg_cluster} -n ${pg_ns}
+      sleep ${wait}
+      count=$((count + 1))
+      continue
+    fi
+  done
+
+  if [ -z ${pg_pod} ]; then
```

Comment:
Without double quotes this will get a syntax error if `pg_pod` is empty.
```suggestion
  if [ -z "${pg_pod}" ]; then
```